src/arch/x86_64/Mir.zig

```diff
@@ -292,6 +293,7 @@ pub const Inst = struct {
         /// Another instruction.
         inst: Index,
         /// A 32-bit immediate value.
+        /// TODO we should add support for 16- and 8-bit immediate values.
```

Comment:
are you sure? it wouldn't save any bytes in the MIR, and we can just check the range of the value when lowering the MIR to machine code.

```suggestion
```